renderdoc/driver/vulkan/wrappers/vk_resource_funcs.cpp

```diff
@@ -880,12 +883,13 @@ VkResult WrappedVulkan::vkFlushMappedMemoryRanges(VkDevice device, uint32_t memR
         if(size == VK_WHOLE_SIZE)
           size = state->mapOffset + state->mapSize - offs;
 
-        GetResourceManager()->MarkMemoryFrameReferenced(GetResID(pMemRanges[i].memory), offs, size,
-                                                        eFrameRef_CompleteWrite);
+        GetResourceManager()->MarkMemoryFrameReferenced(memid, offs, size, eFrameRef_CompleteWrite);
       }
       else
       {
-        GetResourceManager()->MarkDirtyWithWriteReference(memid);
+        // not sure why the above case is completewrite, making this completwrite for consistency
+        GetResourceManager()->MarkResourceFrameReferenced(memid, eFrameRef_CompleteWrite);
```

Comment:
This should only reference as CompleteWrite if the offset and size covers the entirety of the memory region, otherwise it's a PartialWrite.